src/dotnet/AzureAI.Proxy/Program.cs

```diff
@@ -1,30 +1,60 @@
+using Azure.Monitor.OpenTelemetry.AspNetCore;
 using AzureAI.Proxy.ReverseProxy;
 using AzureAI.Proxy.Services;
-using System.Reflection.Metadata.Ecma335;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Yarp.ReverseProxy.Health;
+
 
 var builder = WebApplication.CreateBuilder(args);
 
-builder.Services.AddSingleton<IManagedIdentityService, ManagedIdentityService>();
+//Application Insights
+var instanceId = Environment.GetEnvironmentVariable("CONTAINER_APP_REPLICA_NAME");
+if (instanceId == null)
+{
+    instanceId = "local";
+}
```

Comment:
```suggestion
var instanceId = Environment.GetEnvironmentVariable("CONTAINER_APP_REPLICA_NAME") ?? "local";  
```

Reply:
changed...I don't like these kind of constructs, but fine for this case